providers/hubspot/internal/custom/payload.go

```diff
@@ -0,0 +1,221 @@
+package custom
+
+import (
+	"sync"
+	"time"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/internal/datautils"
+)
+
+func newBatchPayload(fieldDefinitions []common.FieldDefinition) *BatchPayload {
+	payloads := make([]Payload, len(fieldDefinitions))
+
+	for index, definition := range fieldDefinitions {
+		payloads[index] = *newPayload(definition)
+	}
+
+	return &BatchPayload{
+		Inputs: payloads,
+	}
+}
+
+func newPayload(definition common.FieldDefinition) *Payload {
+	return &Payload{
+		Name:            definition.FieldName,
+		Label:           definition.DisplayName,
+		GroupName:       definition.UniqueProperties.HubspotGroupName,
+		Type:            matchType(definition),
+		FieldType:       matchFieldType(definition),
+		Hidden:          false,
+		Description:     definition.Description,
+		FormField:       true,
+		DataSensitivity: "non_sensitive",
+		HasUniqueValue:  definition.Unique,
+		Options:         newOptions(definition),
+	}
+}
+
+func matchType(definition common.FieldDefinition) string {
+	switch definition.ValueType {
+	case common.FieldTypeString:
+		return "string"
+	case common.FieldTypeBoolean:
+		return "bool"
+	case common.FieldTypeDate:
+		return "date"
+	case common.FieldTypeDateTime:
+		return "datetime"
+	case common.FieldTypeSingleSelect, common.FieldTypeMultiSelect:
+		return "enumeration"
+	case common.FieldTypeInt, common.FieldTypeFloat:
+		return "number"
+	default:
+		return "string"
+	}
+}
+
+func matchFieldType(definition common.FieldDefinition) string {
+	switch definition.ValueType {
+	case common.FieldTypeString:
+		return "text"
+	case common.FieldTypeBoolean:
+		return "booleancheckbox"
+	case common.FieldTypeDate:
+		return "date"
+	case common.FieldTypeDateTime:
+		return "datetime"
+	case common.FieldTypeSingleSelect:
+		return "radio"
+	case common.FieldTypeMultiSelect:
+		return "select"
+	case common.FieldTypeInt, common.FieldTypeFloat:
+		return "number"
+	default:
+		return "text"
```

Comment:
```suggestion
func matchFieldType(definition common.FieldDefinition) (string, error) {
	switch definition.ValueType {
	case common.FieldTypeString:
		return "text", nil
	case common.FieldTypeBoolean:
		return "booleancheckbox", nil
	case common.FieldTypeDate:
		return "date", nil
	case common.FieldTypeDateTime:
		return "datetime", nil
	case common.FieldTypeSingleSelect:
		return "radio", nil
	case common.FieldTypeMultiSelect:
		return "select", nil
	case common.FieldTypeInt, common.FieldTypeFloat:
		return "number", nil
	default:
		return "", fmt.Errorf("unrecognized value type: %s", definition.ValueType)
	}
}
```

One thing I noticed in my testing is that this needs to fail loudly. At one point I tried to use a type of "number" (obviously an error) and the system silently decided to use text instead, resulting in a very unintuitive error (Salesforce said something about how text fields weren't allowed to have a precision value - which is true but the root cause is non-obvious).

I'd prefer if this failed more loudly.

Reply:
Good point, I was not sure what to put here. Error will be the cleanest implementation. I will update Salesforce code too.

Reply:
Addressed both connectors:

<img width="643" height="54" alt="image" src="https://github.com/user-attachments/assets/6c73f236-c24b-4ed2-b0a7-858f0f2f7ccf" />

<img width="611" height="50" alt="image" src="https://github.com/user-attachments/assets/b642f17e-232b-4892-9566-31da83e9ea3a" />